event/store.go

```diff
@@ -0,0 +1,60 @@
+package event
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+)
+
+const defaultEventCountLimit = 5
+
+// Store stores events
+type Store struct {
+	*sync.RWMutex
+
+	events map[string][]*Event // taskname => events
+	limit  int
+}
+
+// NewStore returns a new store
+func NewStore() *Store {
+	return &Store{
+		RWMutex: &sync.RWMutex{},
+		events:  make(map[string][]*Event),
+		limit:   defaultEventCountLimit,
+	}
+}
+
+// Add adds an event and manages the limit of number of events stored per task.
+func (s *Store) Add(e *Event) error {
```

Comment:
Can you change the event parameter to pass by copy instead of reference? That way the caller can't unintentionally change event values once it's already added to the store

Reply:
Oh yes! Good point, thank you!

Reply:
honestly there's a lot of this that I probably sprinkled throughout the code too for the convenience of `nil`. i'll try to update them as I see.